src/dft/backends/cufft/commit.cpp

```diff
@@ -138,27 +139,83 @@ class cufft_commit final : public dft::detail::commit_impl<prec, dom> {
         std::array<int, max_supported_dims> n_copy;
         std::copy(config_values.dimensions.begin(), config_values.dimensions.end(), n_copy.data());
         const int rank = static_cast<int>(config_values.dimensions.size());
-        const int istride = static_cast<int>(config_values.input_strides.back());
-        const int ostride = static_cast<int>(config_values.output_strides.back());
-        const int batch = static_cast<int>(config_values.number_of_transforms);
-        const int fwd_dist = static_cast<int>(config_values.fwd_dist);
-        const int bwd_dist = static_cast<int>(config_values.bwd_dist);
-        std::array<int, max_supported_dims> inembed;
-        if (rank == 2) {
-            inembed[1] = config_values.input_strides[1];
+        std::vector<int> inembed = { config_values.input_strides.begin(),
```

Comment:
It is confusing to add the first value (the offset) to `[io]nembed`. If I understand correctly this works because the first value of `[io]nembed` is never read and we later remove the smallest stride. This could use a few comments IMO

Reply:
added a comment